File: UI/wwwroot/Facturacion/FrontModel/Detalle_Factura.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../../WDevCore/WModules/WComponentsTools.js";
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { Tbl_Factura }  from './Tbl_Factura.js'
import { Cat_Producto }  from './Cat_Producto.js'
class Detalle_Factura extends EntityClass {
   constructor(props) {
       super(props, 'EntityFacturacion');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_DetalleFactura;
   /**@type {Number}*/ Cantidad;
   /**@type {Number}*/ Precio_Venta;
   /**@type {Number}*/ Iva;
   /**@type {Number}*/ Total;
   /**@type {Number}*/ Id_Lote;
   /**@type {Number}*/ Descuento;
   /**@type {Number}*/ Sub_Total;
   /**@type {Tbl_Factura} ManyToOne*/ Tbl_Factura;
   /**@type {Cat_Producto} ManyToOne*/ Cat_Producto;
}
export { Detalle_Factura }

File: UI/CAPA_NEGOCIO/Facturacion/Mapping/Detalle_Factura.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
	public class Detalle_Factura : EntityClass
		{
				[PrimaryKey(Identity = true)]
				public int? Id_DetalleFactura { get; set; }
				public int? Id_Factura { get; set; }
				public int? Id_Producto { get; set; }
				public Double? Cantidad { get; set; }
				public Double? Precio_Venta { get; set; }
				public Double? Iva { get; set; }
				public Double? Total { get; set; }
				public int? Id_Lote { get; set; }
				public Double? Descuento { get; set; }
				public Double? Sub_Total { get; set; }				
				public double? Monto_Descuento { get;  set; }
				
				[ManyToOne(TableName = "Tbl_Factura", KeyColumn = "Id_Factura", ForeignKeyColumn = "Id_Factura")]
				public Tbl_Factura? Tbl_Factura { get; set; }
				
				[ManyToOne(TableName = "Tbl_Lotes", KeyColumn = "Id_Lote", ForeignKeyColumn = "Id_Lote")]
				public Tbl_Lotes? Lote { get; set; }
		}
}
